src/optimum/nvidia/hub.py

```diff
@@ -181,17 +229,15 @@ def convert_and_build(
         if engine_config.plugins_config is None:
             engine_config.plugins_config = model_config.get_plugins_config()
 
-        max_memory = get_max_memory()
-
         if hf_model is None:
             LOGGER.debug(
                 f"Loading weights from {local_path} into the model ({cls.HF_LIBRARY_TARGET_MODEL_CLASS.__name__})"
             )
 
             hf_model = cls.HF_LIBRARY_TARGET_MODEL_CLASS.from_pretrained(
                 local_path,
-                device_map="auto",
-                max_memory=max_memory,
+                torch_dtype="auto",
```

Comment:
shouldn't we always load in fp16/bf16? do we ever want to load in fp32? or the fp32 -> lower precision is handled later?

Reply:
I would say we want to load in the dtype the checkpoint is saved. 

`"auto"` will read it from the config and default to something else if not found